🚀[Release v3.4.2] Merge into Main

kcw-grunt opened the pull request:
## Release Notes

### v3.4.2
- Added Punjabi , Polish and Persian
- Fixes

voisine committed:
Initial commit

Commit:
refactor: squash history to single commit

kcw-grunt committed:
Updated gitignore with new filenames

kcw-grunt committed:
Still not finding BRCore

kcw-grunt committed:
update config

kcw-grunt committed:
removed nettle

kcw-grunt committed:
removed the unbound and cleaned

kcw-grunt committed:
moved files to br-swift

kcw-grunt committed:
WIP import BRCore

kcw-grunt committed:
update gitignore

kcw-grunt committed:
removed the TE xcscheme

kcw-grunt committed:
Updated to Xcode 16

kcw-grunt committed:
readdded launch resources

kcw-grunt committed:
remove wrong ver of core

kcw-grunt committed:
remove wrong ver of data

kcw-grunt committed:
STASHING WIP

kcw-grunt committed:
Removed the data submodules

kcw-grunt committed:
WIP...BRCore found but extensions not found

progress

kcw-grunt committed:
DID IT DID!!!!!

Compiled!!!

kcw-grunt committed:
Merge remote-tracking branch 'origin/techdebt/kcw-debugging' into develop

# Conflicts:
#	.gitignore
#	LICENSE

kcw-grunt committed:
debugged the tests

kcw-grunt committed:
updated the .gitignore

kcw-grunt committed:
version bump

kcw-grunt committed:
Removed Today Extension

- Removed nothing test

kcw-grunt committed:
Added localized strings

update references to strings
- Readded BIP39 list
update rough localization strings
Updated the core library
move references to critical files

kcw-grunt committed:
Polished the welcome start view to match Android

kcw-grunt committed:
Set user theme mode globally

kcw-grunt committed:
update brand images

- removed cruft
- updated dark light mode for logo
- removed extra image

kcw-grunt committed:
Implement dark light mode in Start classes

kcw-grunt committed:
Fixed tests

kcw-grunt committed:
Merge pull request #1 from gruntsoftware/feat/update-welcome-view-issue-117

Feat/update welcome view issue 117

kcw-grunt committed:
Replaced UIColor and Color with Brainwallet Color theme (#2)

- Added color assets draft

kcw-grunt committed:
Epic/gtm color ops fixes (#4)

* Replaced UIColor and Color with Brainwallet Color theme (#2)

- Added color assets draft
- Removed the AnnouncesView (for old email campaign)

Updated all backgroundColor = .white

some dead comments

Updated color scheme :

- PinPadViewC
-LoginViewC

Updated colors of StartFlowPresenter

* Send works

- Transactions are being added but not showing up
- Added draft views
- Fixed lock screen colors
- added new images
add new universal icons
Removed old comments
added moonpay logo
Removed extra TabBarVC
Removed extra TabBar VC
removed the iPad idiom
More color replacements
Removed bitrefill
added timestamp to progress view
removed bad tests
Nice . Able to send
----trasnactions are all missing

* More cleanup

WIP Remvoed NoTabBarVC
-polished views
-Added in UIViews
WIP Hacked in new MainView SwiftUI

ADDING BRAND  new Main View  Storyboards are horrible

Go back to fixing the transactions

DEBUGGING Transactions cells

Debuggiing Send View
fixed sendview controller
sent again
Fixed more colors

* Created FooterView

- moved buttons into HostingCOntroller
Bump and added url
Synchonize dark theme
User pref dark set

* version bump

* Moved more elements in the SwiftUI HeaderView

kcw-grunt committed:
chore: ios: Fix smells  comply with App Store (#5)

kcw-grunt committed:
🎥  Epic/gtm refactor core library (#6)

* Found the bug!!!!

Had commented out a bunch of code that was being used
Fixed CustomViewTitle color
fixed-test
version bump
Use brainwallet.co mobile top up
Update WebBuyView.swift
removed comments

* version bump

* Covered the settings colors for light

Just before updating the core
Cruft fixes

* more color fixes

* version bump

* update the core library

* Added Blockheight label

* version bump

kcw-grunt committed:
🧰  Fix/confirm seed words submit (#9)

* Updated core commit

changed Xcode to compile

* Added more tags to debug broken transactions

Added a workaround from Eskimo (iOS 18.4.1 bug)

https://developer.apple.com/forums/thread/777999

Added .ephemeral to the URLSession

Testing to see if iOS 18.3.1 shows the transactions

* version bump

Refactored the submit button
Added some wireframe views
Added dismiss for ConfirmPaperVC

* version bump

- Added test ERROR message
- polish sync label

* Changes the print to debugPrint

kcw-grunt committed:
📽️  Epic/ Add storekit private lib (#7)

* Found the bug!!!!

Had commented out a bunch of code that was being used
Fixed CustomViewTitle color
fixed-test
version bump
Use brainwallet.co mobile top up
Update WebBuyView.swift
removed comments

* version bump

* Added Private StoreKit SwiftPackage

- Cleaned up xcode schemes
-updated to latest core
-updated .gitignore
-Removed unsued code

* Added empty storekit config

Added BrainwalletiOSStoreKite SPM

* updated core

* adjust the sync block height label

* changed to debugPrint

* update gitignore

kcw-grunt committed:
🎥  Epic/add more swift UI views prior to Localizations (#8)

* Send works!

- added rate update
- wireframing the onboarding
- Set basic onboarding
- Added custom dismiss button to ReadyView
- Added more localizations

Fixed the start flow

Core was changed to the wrong commit

Added more assertFailure messages

* version bump

* Ripped Localizations

ALL Broken!!!!!!

* WIP: Added then Localizations Strings catalog

* WIP Added more english text

-Framed in more Views

* update .gitignore

* Re-added the hacked in BuyHostViewController

* version bump

* Added Crashlytics dSym Files per Firebase

https://firebase.google.com/docs/crashlytics/get-started?platform=ios&authuser=0&hl=en#set-up-dsym-uploading

* switched to debugPrint statements

- removed unused swiftformat

* removed commented unused code

* removed unused code

kcw-grunt committed:
🧰  Fix/add localizations (#10)

* Send works!

- added rate update
- wireframing the onboarding
- Set basic onboarding
- Added custom dismiss button to ReadyView
- Added more localizations

Fixed the start flow

Core was changed to the wrong commit

Added more assertFailure messages

* version bump

* Ripped Localizations

ALL Broken!!!!!!

* WIP: Added then Localizations Strings catalog

* WIP Added more english text

-Framed in more Views

* update .gitignore

* Re-added the hacked in BuyHostViewController

* version bump

* Added Crashlytics dSym Files per Firebase

https://firebase.google.com/docs/crashlytics/get-started?platform=ios&authuser=0&hl=en#set-up-dsym-uploading

* Added all AI translated xliff

* Added more Strings(localized tags

* Updated all strings catalogs

Removed old code manageWallet

* version bump

- Removed some exuberant assertFailures

* Current StartView with no Language using the user preferred

* Removed all old Language classes

iOS will auto switch

kcw-grunt committed:
Removed the wipe wallet button (#14)

- Added delete data from the settings
- updated service fees
- Added more strings for localization

kcw-grunt committed:
🧰  Fix: Made Scanning work again (#15)

* Polished the UI

* Changed the PIN Length to 4

* Polished the Pin Lock screen

* Polished the Show My Seed words

* Added localizations fixes

* Resorting file structure (Tidying)

Polishes that are general
Resize the image

kcw-grunt committed:
Add a animated shape for Send (#17)

kcw-grunt committed:
✨ Worked in the Full SwiftUI with LockScreen (#16)

* Worked in the Full SwiftUI with LockScreen

Polish Start
Fixed PinView leading trailing header
99% SwiftUI  LockScreen
Debugging. the PIN AUTH

* Refactor cruft

- Removed audio files
- polished layout

* polished the localizations

* updated the special shape

Updated package

kcw-grunt committed:
per @hassanashraf92 notes (#19)

Refactored opsFees
Added a test

kcw-grunt committed:
Refactored to remove strong parent = child references to avoid 🧟‍♂️ (#18)

kcw-grunt committed:
❇️  chore/fetch moonpay countries+ signed buy url (#20)

* WIP: Routed UI changes

Fetch the info and killed 3 tabs

* Refactored in the Buy/Receive for building

Localizable update

Reintegrated Shapes from other merged branch

* Added shapes

removed some cruft
- refactored
- All Tabs work
- Debugged auto select from old business logic
- Renamed Classes BR to BW
- Adding in Rcv
- Switching from Combine to closure for tap QR as launch was firing the RCV modal

* Buzzed out the buy auth through the view stack

* WIP: Add the API Calls

MVP Level layout complete

* Updated the saving of the currencies ud

Success pulling data from API Server
Debug Call for the buy and limits
add more state vars
Heavy refactor to use BuyQuote

* RC adding signedURL

Remove cruft
Added done set amount button

* Successful signedURL!!

* Used AI Generated ViewModelTests

* version and build number

kcw-grunt committed:
change file location

kcw-grunt committed:
chore: Polish UI (#22)

* version bump

- Update buy and preferred currency across Views to show rates
- Updated localizations
- Polish Segments in Buy view
- Added a MP Logo

* change the symbol

kcw-grunt committed:
build bump

kcw-grunt committed:
fix: ios: Add wipe button UI polishes (#25)

* UI polishes

build number bump
Small polish of start view
Added wipe button
Adjusted the progress bar color

* resolve package

* Removed the prompt cell as a presentation

- reorged files
- Polished the wipe wallet view

* bump build number

kcw-grunt committed:
Added more defensive steps in Wallet Coordinator (#26)

- possible fix
- reshow the rates
- Cover negative amount

More checks

kcw-grunt committed:
Fix/localization polishes (#27)

* Added more defensive steps in Wallet Coordinator

- possible fix
- reshow the rates
- Cover negative amount

More checks

* Set the schemes for Russia and Languages

* Added locale debug label

- Deactivate of the Locale button in settings

* Added debug localization label

build bump

kcw-grunt committed:
Added agent string encryp (#29)

kcw-grunt committed:
🧰   ✨    Chore/currency set defaults polish (#30)

* WIP: Routed UI changes

Fetch the info and killed 3 tabs

* Refactored in the Buy/Receive for building

Localizable update

Reintegrated Shapes from other merged branch

* Added shapes

removed some cruft
- refactored
- All Tabs work
- Debugged auto select from old business logic
- Renamed Classes BR to BW
- Adding in Rcv
- Switching from Combine to closure for tap QR as launch was firing the RCV modal

* Buzzed out the buy auth through the view stack

* WIP: Add the API Calls

MVP Level layout complete

* Updated the saving of the currencies ud

Success pulling data from API Server
Debug Call for the buy and limits
add more state vars
Heavy refactor to use BuyQuote

* RC adding signedURL

Remove cruft
Added done set amount button

* Successful signedURL!!

* Used AI Generated ViewModelTests

* version and build number

* Updated circleci config

* Update Fastfile

Updated the config and fastlane

fixed config remove tab

set ruby version is 3.3.1

Silence or remove chatty warnings

Silence or remove chatty warnings

WIP

* Added swiftlingyml

NO OP .
@hassanashraf92  Can you take a look?

* Updated the README

* WIP swiftlinting

* Moved ReceiveView to its file

* ran swiftlint --fix and successfully ran

Refactor the gitignore
Met linter requirements

* update of the CI config

* Iteration infinity

updated Gemfile
reduced Gem imports
reduced to scan

* Refactor swiftlinter

* Refactor

- Renamed Analytics
- Set ATS
- Removed unused code

* WIP  refactoring the start view

* Cleanup StoreKit

* WIP

* WIP adding the currency update

* Refactor linting

Removed welcome / start  price
Auto lint and small fix

kcw-grunt committed:
build bump

kcw-grunt committed:
✨     Feat/add buy receive inlockscreen (#32)

* Currency polish and wipe steps

Setting query timing of rates call
Refactor remove cruft
Renamed DefaultCurrency to UserPreferredCurrency

* WIP Inside rcv

Working out the layout
reduce modal presenter class
Moved and split the modal monster
good Receive View progress
polished recv cannot buy modal
further polish
worked in the buy receive

* update package

* build bump

kcw-grunt committed:
build bump

build bump
Currency polish and wipe steps (#31)
Setting query timing of rates call
Refactor remove cruft
Renamed DefaultCurrency to UserPreferredCurrency

kcw-grunt committed:
Fix/UI feedback send receive (#35)

* polished the receive views based on feedback

* bump build number

* bump build

kcw-grunt committed:
small polish (#34)

Still debugging the lock screen theme
Rc polish

kcw-grunt committed:
admin push

Polished for swiftlint

kcw-grunt committed:
🧰  Fix Debug price recv (#37)

* Added sync time analytics

- adjusted

* build bump

changed debug label

commented prepped code

* reran swiftlint

xcode re-organation

* buildbump

* Removed the background of Alert color background

migrated rec address
Updated with recv and buy views
Added more tests
polished the modal view

* Fix/UI feedback send receive (#35)

* polished the receive views based on feedback

* bump build number

* bump build

* build bump

build bump
Currency polish and wipe steps (#31)
Setting query timing of rates call
Refactor remove cruft
Renamed DefaultCurrency to UserPreferredCurrency

* Polished for swiftlint

* build bump

set name of the userPreferredDarkTheme

kcw-grunt committed:
🧰   Fix/theme preferred settings (#40)

* Resolved issues found in develop branch

* Reconnected the button in the start view

* Got theme working!!!

Checking any cruft

* Polished the Firebase SDK

- Updated the SDK

* remove simple cruft

* 90% Dark theme polished

* Fixed the background in the main vc

Refactor
Removed code smell
Polished dark and light

* build bump

* fixed tabs in config

kcw-grunt committed:
build a bump

kcw-grunt committed:
build bump

kcw-grunt committed:
Merge branch 'main' into release/v3.4.2

josikie reviewed (dismissed):
Wow! More languages. Thanks @kcw-grunt

kcw-grunt committed:
❇️    Chore/polish localizations (#42)

* Polish the localizations

* Added Punjabi and Persian

code bump
stashed localization coverage at xliff-coverage-check-wip

* Updated missing localizations

* further localization polish

* build bump

* Updated localizations

- Polish
- All others

kcw-grunt committed:
chore: ios: add unique ID to externalid (#44)

* add unique ID to externalid

- There would be collisions without this component

* bump

kcw-grunt commented:
Waiting for App Store Approval

kcw-grunt committed:
build bump

kcw-grunt committed:
Merge branch 'develop' into release/v3.4.2

kcw-grunt committed:
build bump

- linter update
- Polish the send

kcw-grunt commented:
This is now approved. @josikie will you review to approve?  Thanks!

josikie reviewed (approved):
Really? That's awesome 😎

kcw-grunt commented:
Terima Kasih @josikie !
